src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java

```diff
@@ -5,32 +5,124 @@
 
 package com.aws.greengrass.secretmanager;
 
+import com.aws.greengrass.authorization.AuthorizationHandler;
+import com.aws.greengrass.authorization.Permission;
+import com.aws.greengrass.authorization.exceptions.AuthorizationException;
 import com.aws.greengrass.logging.api.Logger;
 import com.aws.greengrass.logging.impl.LogManager;
-import lombok.AccessLevel;
-import lombok.Setter;
+import com.aws.greengrass.secretmanager.exception.v1.GetSecretException;
+import com.aws.greengrass.secretmanager.model.v1.GetSecretValueError;
+import com.aws.greengrass.secretmanager.model.v1.GetSecretValueResult;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import software.amazon.awssdk.aws.greengrass.GeneratedAbstractGetSecretValueOperationHandler;
 import software.amazon.awssdk.aws.greengrass.model.GetSecretValueRequest;
 import software.amazon.awssdk.aws.greengrass.model.GetSecretValueResponse;
+import software.amazon.awssdk.aws.greengrass.model.ResourceNotFoundError;
+import software.amazon.awssdk.aws.greengrass.model.ServiceError;
+import software.amazon.awssdk.aws.greengrass.model.UnauthorizedError;
 import software.amazon.awssdk.eventstreamrpc.OperationContinuationHandlerContext;
 import software.amazon.awssdk.eventstreamrpc.model.EventStreamJsonMessage;
 
+import java.io.IOException;
 import javax.inject.Inject;
 
+import static software.amazon.awssdk.aws.greengrass.GreengrassCoreIPCService.GET_SECRET_VALUE;
+
 /**
  * Class to handle business logic for all SecretManager requests over IPC.
  */
 public class SecretManagerIPCAgent {
     private static final Logger logger = LogManager.getLogger(SecretManagerIPCAgent.class);
+    private static final ObjectMapper OBJECT_MAPPER =
```

Comment:
~~can we reuse an object mapper provided by nucleus? i forget where it is off the top of my head.  we should also have object mapper allow unknown properties, which allows backwards compatibility if new fields are added~~

Reply:
ah nevermind, there's already one in `SecretManagerService`, lets have both places referring to the same mapper then

Reply:
Removed it.